sdk/storage/Azure.Storage.Common/src/Shared/StreamSlice.cs

```diff
@@ -0,0 +1,16 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+namespace Azure.Storage.Shared
+{
+    /// <summary>
+    /// Describes a stream that is a partition of another, larger stream.
+    /// </summary>
+    internal abstract class StreamSlice : System.IO.Stream
```

Comment:
Since WindowStream is now abstract I think we could get rid of this type and just use WindowStream instead where AbsolutePolistion is needed. 
Btw. "StreamSlice" and "WindowStream" seem to be same thing conceptually (naming wise).

Reply:
I was discussing this with Sean the other day regarding `PooledMemoryStream` and his use of it for OpenWrite. Unfortunately, since both this and his PRs touch the same code, it would not be fun for either of us to make the appropriate refactors until we've both merged. #12820 was filed to address this regarding `PooledMemoryStream`, and we can address `WindowStream` along with it.

Reply:
Ok.